src/System.Diagnostics.TraceSource/src/System/Diagnostics/TraceInternal.cs

```diff
@@ -369,6 +340,7 @@ internal static void Refresh()
             {
                 s_settingsInitialized = false;
                 s_listeners = null;
+                SetIndentSize(DiagnosticsConfiguration.IndentSize);
```

Comment:
Fixes StackOverflowException due to OnIndentLevelChanged/OnIndentSizeChanged refactoring without changing behavior